Databricks mcp scaffolding

FlagBenett opened the pull request:
## Description

Scaffolding the databricks mcp in front
- behind a feature flag
- basic architecture and 1 tool (list warehouses)
- personal or shared oauth
- Oauth provider including client_id, client_secret and databricks_workspace

## Tests

locally

## Risk

low

## Deploy Plan

front only

FlagBenett committed:
databricks oauth provider and addition in sdk for front

FlagBenett committed:
databricks front scaffolding

FlagBenett committed:
add mcp server

FlagBenett committed:
merge main

FlagBenett committed:
Merge branch 'main' into databricks-mcp-scaffolding

FlagBenett committed:
lint

FlagBenett committed:
bump server id

aubin-tchoi commented on front/lib/actions/mcp_internal_actions/servers/databricks/databricks_api_helper.ts:
```diff
@@ -0,0 +1,144 @@
+  } catch (error: unknown) {
+    logger.error("Error in withAuth", { error });
+    return new Err(
+      new MCPError(`Authentication error: ${normalizeError(error).message}`)
```
Authentication error is untrue, we are catching everything here

aubin-tchoi commented on front/lib/actions/mcp_internal_actions/servers/databricks/databricks_api_helper.ts:
```diff
@@ -0,0 +1,144 @@
+    );
+  }
+
+  try {
```
let's remove this try catch

aubin-tchoi commented on front/lib/actions/mcp_internal_actions/servers/databricks/rendering.ts:
```diff
@@ -0,0 +1,12 @@
+import type { Warehouse } from "./types";
```
relative import

FlagBenett committed:
remove trycatch

FlagBenett committed:
remove 2nd trycatch

FlagBenett committed:
use function keyword

FlagBenett committed:
Merge branch 'main' into databricks-mcp-scaffolding
	# Please enter a commit message to explain why this merge is necessary,
		# especially if it merges an updated upstream into a topic branch.
		#

FlagBenett committed:
use typeguard and absolute imports

FlagBenett committed:
use typeguard

FlagBenett commented on front/lib/api/oauth/providers/databricks.ts:
```diff
@@ -0,0 +1,200 @@
+    return Object.keys(extraConfig).length === 0;
+  }
+
+  async getRelatedCredential(
```
yes, similarly to salesforce, we need admins to create their oauth app in databricks, and save their client id and secret in Dust

aubin-tchoi commented on front/lib/actions/mcp_internal_actions/servers/databricks/index.ts:
```diff
@@ -0,0 +1,76 @@
+
+              if (result.isErr()) {
+                return new Err(
+                  new MCPError(result.error.message ?? "Unknown error")
```
Small detail but I think you already have an MCPError and can you return result here

aubin-tchoi reviewed (approved):
LGTM 🔥

FlagBenett committed:
remove try catch and update error type to MCPError

FlagBenett committed:
use isString() in typeguard
